### DISCUSSION TITLE
Where to install Navigation Guards. Before Enter, Before Each, and After Each

### DISCUSSION THREAD

**Discussion by TomT15** [posva/unplugin-vue-router#406] [Questions and Help] [answered]
Right now I use the three navigation guards and since I'm trying to use the router/auto-routes im not sure how to implement them. 

Originally they were added as such
```
{
          path: '/dealer',
          name: 'DealerDashboard',
          beforeEnter: (to, from) => {
            const jwtStore = useJWT();
            if (jwtStore.tokenPayload.Role === 'Admin') {
              return true
            };
            return { name: "UserDashboard", path: "/UserDashboard" }
},
```
but after some testing i learned if i add anything to my router/index.ts file it would completely break out how router/auto worked.

If possible a small example and explaining on where this code will go would be very appreciated. Thank you

**Comment by posva** [comment, marked as answer]
- `beforeEnter()`: https://uvr.esm.is/guide/extending-routes.html#definepage
- `beforeEach()` and `afterEach()` goes where you define the router instance

**Reply by TomT15** [reply]
Thank you for clarifying that.

I had tried some things out and ran into issues so originally I thought i was doing it wrong. Right now im having issues trying to access my store in the beforeEnter

```
definePage({
  beforeEnter: () => {
    const jwtStore = useJWT();
    if (jwtStore.tokenPayload?.Role === 'Admin') {
      return true;
    };
    return { name: "DealerDashboard", path: "/DealerDashboard" }
  }
})
```

Do you have any suggestions?

Right now its just saying **useJWT is not defined**

i did double check and make sure all the necessary imports were there.

**Reply by posva** [reply]
It’s not supported yet https://github.com/posva/unplugin-vue-router/issues/317
Although you could use a dynamic import within the beforeEnter and that would work

**Reply by userquin** [reply]
@posva how about using another script with options api? should it work? `script + script setup`  with `beforeEach()` and `afterEach()` guards in the script (I used this approach with vue 2 + antfu vue 2 script setup lib and is working)

![imagen](https://github.com/posva/unplugin-vue-router/assets/6311119/11a9dbc6-ebee-499c-95d0-f036885597d9)

**Reply by posva** [reply]
That works too
